### DISCUSSION TITLE
How to adjust page width

### DISCUSSION THREAD

**Discussion by micartey** [withastro/starlight#2180] [Q&A] [answered]
### What version of `starlight` are you using?

0.25.4

### What is your idea?

The page width (the part where I write the documentation) appears to be rather narrow. Especially on wide screen monitors.

### Why is this feature necessary?

This will improve the look of the documentation on wide screen monitors 

### Do you have examples of this feature in other projects?

![image](https://github.com/user-attachments/assets/b72533dd-ab74-44db-9916-cead9ecc11f3)

(Current)

![image](https://github.com/user-attachments/assets/64f6afd3-8960-44a2-8108-806cb4708755)

(The way it should look)

### Participation

- [X] I am willing to submit a pull request for this feature.

**Comment by micartey** [comment]
Please note that I edited the css that has been generated by a production side. This is not perfect by any means and should only illustrate what I mean

**Comment by delucis** [comment, marked as answer]
Hey @micartey! The default column width is chosen to have a reasonable line length (a.k.a measure) for comfortable reading of body text. In other words, it’s optimised such that when a reader needs to scan back from the end of one line in a paragraph to the start of the next, the distance is comfortable enough for the reading experience to flow. (This is always a balancing act with typography — too narrow and these jumps are too frequent, too wide and these jumps are harder for the eye to track.)

In your screenshot the content is admittedly mostly not text, and there are certainly examples like this that can benefit from a wider width.

This is supported by setting the `--sl-content-width` custom property in your CSS. For example, on https://starlight-examples.netlify.app/ I increased Starlight’s default `45rem` to `70rem` to better accommodate the large embeds on its pages:

```css
:root {
	--sl-content-width: 70rem;
}
```

Even then, I still restricted body text to not use all that space to avoid very long lines that are uncomfortable to read (see https://starlight-examples.netlify.app/examples/custom-title-link/ for an example):

```css
.sl-markdown-content :is(p, li) {
	max-width: 45rem;
}
```

Depending on your content, that might be too simple and some more styles are needed to handle different circumstances, but hopefully that can point you in the right direction!

**Reply by micartey** [reply]
Thank you this helps a lot!
